src/ray/raylet/node_manager.cc

```diff
@@ -326,6 +326,30 @@ void NodeManager::Heartbeat() {
     heartbeat_data->add_resource_load_capacity(resource_pair.second);
   }
 
+  std::unordered_set<ObjectID> active_object_ids = worker_pool_.GetActiveObjectIDs();
```

Comment:
We should also consider the arguments of tasks that are queued in the raylet. Maybe this can be done in a subsequent PR.

Reply:
Good point. I think this will add non-trivial complexity to the PR so let's do it in a follow up.